lxc/cookiejar/jar.go

```diff
@@ -16,29 +18,28 @@ import (
 )
 
 // Open opens a new jar at the given location, reading contents from disk if present.
-func Open(filepath string) (*Jar, error) {
-	jar, err := newJar(filepath)
+func Open(filepath string, host string) (*Jar, error) {
+	jar, err := newJar(filepath, host)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to open jar: %w", err)
+		return nil, fmt.Errorf("failed to open jar: %w", err)
 	}
-
-	err = jar.openJar()
-	if err != nil {
-		return nil, fmt.Errorf("Failed to open jar: %w", err)
+	if err := jar.openJar(); err != nil {
+		return nil, fmt.Errorf("failed to open jar: %w", err)
 	}
-
 	return jar, nil
 }
 
 // newJar creates a new [Jar] but does not attempt to read existing contents of the jar.
-func newJar(filepath string) (*Jar, error) {
-	// Use the default public suffix list.
+func newJar(filepath string, remote string) (*Jar, error) {
 	jar, err := cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
 	if err != nil {
-		return nil, fmt.Errorf("Failed to instantiate a cookie jar: %w", err)
+		return nil, fmt.Errorf("failed to instantiate a cookie jar: %w", err)
 	}
-
-	return &Jar{Jar: jar, filepath: filepath}, nil
+	remoteURL, err := url.Parse(remote)
```

Comment:
Please add a newline after each functional block. E.g.:
```go
/* logic */
if err != nil {
    /* handle err */
}

remoteURL, err := url.Parse(remote)
```

We have a linter for this, as it generally improves readability.